internal/provider/provider_test.go

```diff
@@ -113,7 +108,9 @@ func testConfig(res ...string) string {
 	return strings.Join(c, "\n")
 }
 
-func TestRead(t *testing.T) {
+// If project ID is not defined on the provider config, the provider
+//project ID becomes the organization's oldest existing project
+func TestDetermineOldestProject(t *testing.T) {
```

Comment:
I think the provider/provider_test files are getting a little bogged down with the project helpers and tests. Can we move the project helpers and their tests to `project.go` and `project_test.go`?